tests/tests_ansible.yml

```diff
@@ -375,47 +375,47 @@
         register: result
         failed_when: result is failed or result is changed
 
-
-    always:
-
-      # CLEANUP: RESET TO ZONE DEFAULTS
-
-      - name: Remove custom zone
-        command: firewall-cmd --permanent --delete-zone=custom
-        register: result
-        failed_when: result.failed and "INVALID_ZONE" not in result.stderr
-
-      - name: Reset internal zone to defaults
-        command: firewall-cmd --permanent --load-zone-defaults=internal
-        register: result
-        failed_when: result.failed and "NO_DEFAULTS" not in result.stderr
-
-      - name: Reset trusted zone to defaults
-        command: firewall-cmd --permanent --load-zone-defaults=trusted
-        register: result
-        failed_when: result.failed and "NO_DEFAULTS" not in result.stderr
-
-      - name: Reset dmz zone to defaults
-        command: firewall-cmd --permanent --load-zone-defaults=dmz
+      - name: set previous replaced
+        firewall_lib:
+          zone: dmz
+          state: enabled
+          icmp_block_inversion: yes
+          permanent: yes
+          immediate: yes
+          previous: replaced
```

Comment:
The issue here with custom zones is that `previous: replaced` is processed first.  This erases everything, including the custom zone.  In order to add a zone, you must pass a special zone add argument list like this:
```
      - name: set previous replaced
        firewall_lib:
          zone: customzone
          state: present
          permanent: yes
          immediate: yes
          previous: replaced
```
Note `present` instead of `enabled`, and `zone` is the only object being set.

In the role interface (as opposed to this which is the module interface), the user will be passing in a list, so they can do something like this:
```
firewall:
  - previous: replaced
    zone: customzone
    state: present
    permanent: yes
    immediate: yes
  - zone: customzone
    state: enabled
    icmp_block_inversion: yes
    permanent: yes
    immediate: yes
```
that is - two operations - the first one erases everything then adds the customzone, the second adds icmp_block_inversion to the customzone

which brings up another problem - even though the *module* interface is idempotent, the *role* interface is not.  Consider this - if you do this:
```
include_role:
  name: linux-system-roles.firewall
vars:
  - previous: replaced
    zone: customzone
    state: present
    permanent: yes
    immediate: yes
  - zone: customzone
    state: enabled
    icmp_block_inversion: yes
    permanent: yes
    immediate: yes
```
This will return changed=true every time you run it, because `previous: replaced` will always remove the customzone and its settings.  So the role is not idempotent.